### ISSUE TITLE
Extending the current documentation and wiki

### ISSUE THREAD

**Issue by dodoradio** [AsteroidOS/asteroidos.org#237]
frequently, conversation comes up about documenting certain specifics about individual smartwatches. I personally agree with this, since it is difficult to find robust information on the internet, with certain watches only being covered in reviews that do not cover any technical aspects. 

This would be quite a large addition to our documentation, so it is worth considering how we want to take this forward. 

A few methods have been suggested to allow these pages to be added:
- extend the current `Documentation` section on the website. This has the advantage of fitting comfortably into our current build system, and should give full control over the entire appearance and behaviour of the wiki pages, while providing a familiar markdown interface for implementing the content. The disadvantages are that 1) the documentation is difficult and unintuitive to edit, requiring use of github and standing in the way of spontaneous corrections or small additions. 2) beyond the most basic level of creating text-and-image articles, our current system is quite difficult to use. Any additions beyond the current templates require experience with javascript and html, and testing requires a local version of the website to be set up. While the website has been containerised, this setup is still quite a high bar for entry. 
- add an instance of a common wiki system, such as mediawiki. The main advantages come with on-the-fly what-you-see-is-what-you-get editing that doesn't require the use of git or a local website setup for editing, while using wiki definition languages common across many projects' documentation pages. There are also a few disadvantages: 1) mediawiki will never, without unreasonable time investment, have a look consistent with the main website. There are a lot of theming options that are not visually unappealing, but consistency is not really achievable.. 2) moderation becomes a far more serious problem than on github. Tampering will need to be fixed after the fact, instead of being stopped at the pull request stage. 3) mediawiki is very powerful, but some functionality can be considered as bloat.

I think that the wiki paradigm we choose now will be something that will be using for a very long time, so it is best to choose something that provides a good balance of customisation, functionality and extensibility

**Comment by eLtMosen** [comment]
Having discussed with @beroset . He mentioned we could maybe convert the whole website into a wiki.
And i think that is a tempting idea.
If we implement a wiki into the current website. It will be the largest piece of content anyway.
The documentation, install and FAQ pages could easily be a wiki.
Leaving 5 pages that make up the website being not wiki related.

But imho one strong point of our current website are the auto generated install pages.
How would we implement such things into a wiki, usual wikis would result in much text duplication?
Ideally, if we touch everything, can we have sort of a watch database where we generate the articles from and maybe have a decent search feature for watch spec details?

**Comment by beroset** [comment]
Here are some ideas for requirements for however we do it.  Note that the numbering is solely for convenience in talking about individual points and is not intended to reflect any kind of prioritization.

## Proposed requirements
1. visually coherent and pleasing design
2. allow arbitrary hyperlinks to both internal pages and external sites
3. provide some clues for navigation (e.g. current sidenav and top menu)
4. allow for text, pictures, movies integration
5. support data-driven page creation/maintenance (e.g. our current json to install pages thing)
6. support some means for a user, disconnected from the internet, to use a local clone (as with current containerization)
7. provide reasonably good security
8. support customization
9. access control and logging to know who changed what and when
10. support review/rollback of suggested edits
11. allow users to be able to add/correct content without being programmers or git experts
12. tools/technologies should be long-lasting and well supported
13. all tooling is FOSS
14. enable possible future web extensions such as watch-face creation tool
15. provide a clear and rational architecture
16. enable/support/provide a unit test framework for customizations
17. serve pages that look good on computers, tablets, phones  (and maybe watches at some point?)
18. support internationalization
19. enable/support/provide access logging 
20. enable/support/provide error logging

**Comment by FlorentRevest** [comment]
> 1. visually coherent and pleasing design

Yup. Note that if we find *a* wiki which doesn't look too trashy, I'd be happy to have *everything* be a wiki.

> 2. allow arbitrary hyperlinks to both internal pages and external sites

We could call it the web.

> 3. provide some clues for navigation (e.g. current sidenav and top menu)

I think they are nice yeah.

> 4. allow for text, pictures, movies integration
> 5. support data-driven page creation/maintenance (e.g. our current json to install pages thing)

In the worst case scenario we could have scripts that update pages based on checked-in json configs. Also as we step towards more and more detailed info (like your kernel version PR generated out of an awk query) it could also be good to dynamically generate these jsons.

No need to over-engineer things  if there's no need but I'm saying, we could probably do various things.

> 6. support some means for a user, disconnected from the internet, to use a local clone (as with current containerization)

If the wiki system we choose is as good as it promises, I don't think this would be super important.

> 7. provide reasonably good security
> 8. support customization
> 9. access control and logging to know who changed what and when

Probably some amount of built-in spam filtering would be good yes.

> 10. support review/rollback of suggested edits
> 11. allow users to be able to add/correct content without being programmers or git experts
> 12. tools/technologies should be long-lasting and well supported
> 13. all tooling is FOSS
> 14. enable possible future web extensions such as watch-face creation tool

I wouldn't stress too much this one. We already have "fake" extensions to the website like https://log.asteroidos.org/ 

> 15. provide a clear and rational architecture
> 16. enable/support/provide a unit test framework for customizations
> 17. serve pages that look good on computers, tablets, phones  (and maybe watches at some point?)
> 18. support internationalization
> 19. enable/support/provide access logging
> 20. enable/support/provide error logging

**Comment by FlorentRevest** [comment]
Fwiw, dodo did some more lobbying on the topic of getting a wiki and asked me to bump this thread with my latest comments on the aesthetic of the wiki, so, quoted from our chat:

> Re theming, I'm fairly sure that if we keep the theming for later, it will just never happen. But I suppose that if others are fine with this then so be it. I also don't think we should necessarily block something useful on forcing someone into writing a chunk of CSS first (surely the Geneva convention must say something against coercing people into writing CSS)[...] Essentially I feel rather neutral about all of this. I don't have the motivation to either fight for nor against it at the moment and I also don't have the time to help make it happen so whatever happens will probably be fine by me and I'm happy with whatever the other maintainers say

**Comment by beroset** [comment]
Almost all of the proposed requirements have been met and we now have a functioning wiki:  https://wiki.asteroidos.org/index.php/Documentation
